File: src/tournoi/dto/update-tournoi.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import {
  IsString,
  IsOptional,
  IsDateString,
  IsEnum,
  IsNumber,
  Min,
} from 'class-validator';
import { TournoiEtat } from '../interfaces/tournoi-etat.enum';
import { TournoiNiveau } from '../interfaces/tournoi-niveau.enum';

export class UpdateTournoiDto {
  @ApiProperty({
    example: 'Tournoi de Printemps U12',
    description: 'Nom du tournoi',
    required: false,
  })
  @IsOptional()
  @IsString()
  nom?: string;

  @ApiProperty({
    example: 'Tournoi amical pour les jeunes joueurs de la région',
    description: 'Détails sur le tournoi, son objectif, son déroulement',
    required: false,
  })
  @IsOptional()
  @IsString()
  description?: string;

  @ApiProperty({
    example: 'football',
    description: 'Type de sport (football, basketball, natation, etc.)',
    required: false,
  })
  @IsOptional()
  @IsString()
  sport?: string;

  @ApiProperty({
    example: 'U12',
    description: 'Tranche d\'âge concernée (ex: U10, U12, U14)',
    required: false,
  })
  @IsOptional()
  @IsString()
  categorieAge?: string;

  @ApiProperty({
    example: '2024-05-15T09:00:00.000Z',
    description: 'Date de début du tournoi',
    type: String,
    format: 'date-time',
    required: false,
  })
  @IsOptional()
  @IsDateString()
  dateDebut?: string;

  @ApiProperty({
    example: '2024-05-17T18:00:00.000Z',
    description: 'Date de fin du tournoi',
    type: String,
    format: 'date-time',
    required: false,
  })
  @IsOptional()
  @IsDateString()
  dateFin?: string;

  @ApiProperty({
    example: 'Stade Municipal de la Ville',
    description: 'Lieu où se déroule le tournoi (stade, gymnase, etc.)',
    required: false,
  })
  @IsOptional()
  @IsString()
  lieu?: string;

  @ApiProperty({
    example: 16,
    description: 'Nombre maximal de participants ou d\'équipes',
    required: false,
    minimum: 1,
  })
  @IsOptional()
  @IsNumber()
  @Min(1)
  nombreParticipantsMax?: number;

  @ApiProperty({
    example: 25.50,
    description: 'Frais d\'inscription (si applicable)',
    required: false,
    minimum: 0,
  })
  @IsOptional()
  @IsNumber()
  @Min(0)
  fraisParticipation?: number;


  @ApiProperty({
    example: 'ouvert',
    description: 'Statut du tournoi',
    enum: TournoiEtat,
    enumName: 'TournoiEtat',
    required: false,
  })
  @IsOptional()
  @IsEnum(TournoiEtat)
  etat?: TournoiEtat;

  @ApiProperty({
    example: 'intermédiaire',
    description: 'Niveau du tournoi',
    enum: TournoiNiveau,
    enumName: 'TournoiNiveau',
    required: false,
  })
  @IsOptional()
  @IsEnum(TournoiNiveau)
  niveau?: TournoiNiveau;

  @ApiProperty({
    example: 'Trophée et médailles pour les 3 premiers',
    description: 'Description de la récompense pour les gagnants',
    required: false,
  })
  @IsOptional()
  @IsString()
  recompense?: string;
}

File: app/src/main/java/com/example/dam_front/models/UpdateTournoiDto.kt
package com.example.dam_front.models

import com.google.gson.annotations.SerializedName

/**
 * DTO pour la mise à jour d'un tournoi
 * Correspond exactement au UpdateTournoiDto du backend
 * Ne contient QUE les champs autorisés par le backend
 */
data class UpdateTournoiDto(
    @SerializedName("nom")
    val nom: String? = null,
    
    @SerializedName("description")
    val description: String? = null,
    
    @SerializedName("sport")
    val sport: String? = null,
    
    @SerializedName("categorieAge")
    val categorieAge: String? = null,
    
    @SerializedName("dateDebut")
    val dateDebut: String? = null,
    
    @SerializedName("dateFin")
    val dateFin: String? = null,
    
    @SerializedName("lieu")
    val lieu: String? = null,
    
    @SerializedName("nombreParticipantsMax")
    val nombreParticipantsMax: Int? = null,
    
    @SerializedName("fraisParticipation")
    val fraisParticipation: Double? = null,
    
    @SerializedName("etat")
    val etat: String? = null,
    
    @SerializedName("niveau")
    val niveau: String? = null,
    
    @SerializedName("recompense")
    val recompense: String? = null
)









